src/crimson/osd/osd.cc

```diff
@@ -1402,6 +1403,93 @@ seastar::future<> OSD::send_beacon()
   return monc->send_message(std::move(beacon));
 }
 
+seastar::future<> OSD::run_bench(int64_t count, int64_t bsize, int64_t osize, int64_t onum, double* elapsed) {
+  LOG_PREFIX(OSD::run_bench);
+  DEBUG(" count={}, bsize={}, osize={}, onum={}", count, bsize, osize, onum);
+
+  return seastar::do_with(ceph::os::Transaction(), ceph::os::Transaction(),
+    [=, this](auto& t, auto& cleanup) {
+    return seastar::do_for_each(boost::make_counting_iterator<int64_t>(0),
+      boost::make_counting_iterator<int64_t>(onum),
+      [&t, &cleanup, osize](int64_t i) {
+        std::string data(osize, 'a');
+        ceph::buffer::list bl(osize);
+        bl.append(data);
+        std::string oid_str = fmt::format("disk_bw_test_{}", i);
+        ghobject_t oid(hobject_t(sobject_t(object_t(oid_str), 0)),
+          ghobject_t::NO_GEN,
+          shard_id_t::NO_SHARD);
+        t.write(coll_t::meta(), oid, 0, data.size(), bl);
+        cleanup.remove(coll_t::meta(), oid);
+        return seastar::make_ready_future<>();
```

Comment:
~In order to simplify this function, the transaction creation logic can be voided (aka regular for loop). Only the submission itself should be futurized.~
However, when looking again, it might be more correct and correlated with Classic's bench test if we'll submit each write with a separate transaction - and later on we could await all of them using `seastar::when_all_succeed`.
For cleanup we can use a single transaction (as used here).